projects/plugins/jetpack/_inc/lib/markdown/gfm.php

```diff
@@ -240,7 +240,7 @@ protected function shortcode_preserve( $text ) {
 	 */
 	protected function urls_preserve( $text ) {
 		$text = preg_replace_callback(
-			'#(?<!<)(?:https?|ftp)://([^\s<>"\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i',
+			'#(?<!<)(?:https?|ftp)://([^\s<>"\'\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i',
```

Comment:
Technically I wonder if there is a problem here -- I think apostrophes are actually supposed to be allowed in URLs (but double quotes aren't).  In practice I'm not sure how much it matters, because URLs are often fully-encoded, but they don't have to be.

For example, using this script to compare the original and new regexp here:

```
<?php

$url = "https://www.example.com/directory?q='test'";

print "Original regexp, URL with single quotes:\n\n";
print "INPUT:  $url";
preg_match( '#(?<!<)(?:https?|ftp)://([^\s<>"\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i', $url, $matches );
print "\nRESULT: " . $matches[0] . "\n";

print "\nNew regexp, URL with single quotes:\n\n";
print "INPUT:  $url";
preg_match( '#(?<!<)(?:https?|ftp)://([^\s<>"\'\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i', $url, $matches );
print "\nRESULT: " . $matches[0] . "\n";

$url = 'https://www.example.com/directory?q="test"';

print "\nOriginal regexp, URL with double quotes:\n\n";
print "INPUT:  $url";
preg_match( '#(?<!<)(?:https?|ftp)://([^\s<>"\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i', $url, $matches );
print "\nRESULT: " . $matches[0] . "\n";

print "\nNew regexp, URL with double quotes:\n\n";
print "INPUT:  $url";
preg_match( '#(?<!<)(?:https?|ftp)://([^\s<>"\'\[\]()]+|\[(?1)*+\]|\((?1)*+\))+(?<![_*.?])#i', $url, $matches );
print "\nRESULT: " . $matches[0] . "\n";
```

Gives this result:

```
Original regexp, URL with single quotes:

INPUT:  https://www.example.com/directory?q='test'
RESULT: https://www.example.com/directory?q='test'

New regexp, URL with single quotes:

INPUT:  https://www.example.com/directory?q='test'
RESULT: https://www.example.com/directory?q=

Original regexp, URL with double quotes:

INPUT:  https://www.example.com/directory?q="test"
RESULT: https://www.example.com/directory?q=

New regexp, URL with double quotes:

INPUT:  https://www.example.com/directory?q="test"
RESULT: https://www.example.com/directory?q=
```

If we have to deal with that, it's probably going to require a more complicated fix to this regexp than I have time to figure out :smiley: